app/frontend/wallet/cardano-ledger-crypto-provider.js

```diff
@@ -0,0 +1,124 @@
+const TransportU2F = require('@ledgerhq/hw-transport-u2f').default // for browser
+const Ledger = require('ledgerhq/hw-app-ada').default // temporary hack, should be @ledgerhq
+const CachedDeriveXpubFactory = require('./helpers/CachedDeriveXpubFactory')
+const cbor = require('borc')
+const {TxWitness, SignedTransactionStructured} = require('./transaction')
+
+const CardanoLedgerCryptoProvider = async (ADALITE_CONFIG, walletState) => {
+  const transport = await TransportU2F.create()
+  transport.setExchangeTimeout(15 * 60000) // 60000ms = 1minute
+  const ledger = new Ledger(transport)
+
+  const state = Object.assign(walletState, {
+    rootHdPassphrase: null,
+    derivedAddresses: {},
+  })
+
+  if (state.derivationScheme.type !== 'v2') {
+    throw new Error(`Unsupported derivation scheme: ${state.derivationScheme.type}`)
+  }
+
+  const isHwWallet = () => true
+  const getHwWalletName = () => 'Ledger'
+
+  const deriveXpub = CachedDeriveXpubFactory(state.derivationScheme, async (absDerivationPath) => {
+    try {
+      const response = await ledger.getExtendedPublicKey(absDerivationPath)
+      return Buffer.from(response.publicKeyHex.concat(response.chainCodeHex), 'hex')
```

Comment:
Personally, I am not big fan of "concat" call as it looks asymmetric.
Why not `const xpubHex = response.publicKeyHex + response.chainCodeHex` and then `Buffer.from(xpubHex)` ?